bundles/action/org.openhab.action.pushsafer/src/main/java/org/openhab/action/pushsafer/internal/Pushsafer.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.action.pushsafer.internal;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Arrays;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.openhab.core.scriptengine.action.ActionDoc;
+import org.openhab.core.scriptengine.action.ParamDoc;
+import org.openhab.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class contains the methods that are made available in scripts and rules
+ * for sending messages via the Pushsafer mobile device push service..
+ *
+ * @author Chris Graham / Kevin Siml
+ * @since 1.5.0
+ */
+public class Pushsafer {
+
+    private static final Logger logger = LoggerFactory.getLogger(Pushsafer.class);
+
+    private final static String API_URL = "https://www.pushsafer.com/api";
+    private final static String CONTENT_TYPE = "application/x-www-form-urlencoded";
+    private final static String UTF_8_ENCODING = "UTF-8";
+
+    private final static String API_RETURN_ROOT_TAG = "hash";
+    private final static String API_RETURN_STATUS_TAG = "status";
+    private final static String API_RETURN_ERROR_TAG = "error";
+    private final static String API_RETURN_STATUS_SUCCESS = ":1,";
+    private final static int API_MAX_MESSAGE_LENGTH = 4096;
+
+    public static final String MESSAGE_KEY_API_KEY = "k";
+    public static final String MESSAGE_KEY_MESSAGE = "m";
+    public static final String MESSAGE_KEY_DEVICE = "d";
+    public static final String MESSAGE_KEY_TITLE = "t";
+    public static final String MESSAGE_KEY_ICON = "i";
+    public static final String MESSAGE_KEY_VIBRATION = "v";
+    public static final String MESSAGE_KEY_SOUND = "s";
+
+    static int timeout = 10000;
+
+    // Primary method for sending a message to the Pushsafer API
+    @
+    ActionDoc(text = "Send a notification to your iOS, Android or Win10 device. Private or Alias Key and message are required. All other can effectively be null. Check the Pushsafer.com API for more informations: https://www.pushsafer.com/en/pushapi", returns = "<code>true</code>, if successful and <code>false</code> otherwise.")
```

Comment:
"All other can effectively" should say "All others can". (and drop "effectively")
"for more informations" should say "for more information"